tests/unit/source_code/test_known.py

```diff
@@ -73,11 +74,11 @@ def test_error_on_missing_known_json() -> None:
         KnownList._get_known()  # pylint: disable=protected-access
 
 
-@pytest.mark.skip(reason="Interferes with tests reading known list at the same time of rebuilding (writing)")
-def test_rebuild_trivial() -> None:
-    # No-op: the known.json file is already up-to-date
-    cwd = Path.cwd()
-    KnownList.rebuild(cwd)
```

Comment:
I've been thinking about this, and was thinking along the lines of:
a) The `.rebuild()` method gets an extra "update_known_json" parameter that defaults to `False`.
b) The `main()` method that calls it passes in `True`.
c) The branch for updating the file raises an exception if the distributions are different and it's not allowed to write. This would be what the test encounters.

What do you think?

Reply:
I just realised these changes don't fix the problem: by the time we fail _this_ test the `known.json` has already been rewritten, and other (concurrent) tests may also have failed because they read the file while it was being updated.

Given that a test shouldn't be modifying anything in the source tree, I think this needs to be addressed.